pkg/registry/batch/job/storage/storage.go

```diff
@@ -91,6 +102,30 @@ func (r *REST) Categories() []string {
 	return []string{"all"}
 }
 
+func (r *REST) Delete(ctx context.Context, name string, deleteValidation rest.ValidateObjectFunc, options *metav1.DeleteOptions) (runtime.Object, bool, error) {
+	if options != nil && options.PropagationPolicy == nil && options.OrphanDependents == nil {
```

Comment:
limit the warning to requests which default to OrphanDependents via batch/v1 by adding this:

`&& job.Strategy.DefaultGarbageCollectionPolicy(ctx) == rest.OrphanDependents`